test/functional/ui/highlight_spec.lua

```diff
@@ -194,3 +194,60 @@ describe('Default highlight groups', function()
     feed('<cr>') --  skip the "Press ENTER..." state or tests will hang
   end)
 end)
+
+describe('New special highlight color attribute', function() 
+  local screen
+
+  before_each(function()
+    clear()
+    screen = Screen.new(25,10)
+    screen:attach()
+    screen:set_default_attr_ignore({
+      {bold=true, foreground=Screen.colors.Blue}
+    })
+  end)
+
+  it('used for undercurls', function()
+    execute('syntax on')
+    execute('syn keyword TmpKeyword neovim')
+    execute('syn keyword TmpKeyword1 special')
+    execute('syn keyword TmpKeyword2 specialwithbg')
+    execute('syn keyword TmpKeyword3 specialwithfg')
+    execute('hi! Awesome guifg=red guibg=yellow guisp=red')
+    execute('hi! Awesome1 guisp=red')
+    execute('hi! Awesome2 guibg=yellow guisp=red')
+    execute('hi! Awesome3 guifg=red guisp=red')
+    execute('hi link TmpKeyword Awesome')
+    execute('hi link TmpKeyword1 Awesome1')
+    execute('hi link TmpKeyword2 Awesome2')
+    execute('hi link TmpKeyword3 Awesome3')
+    insert([[
+      neovim
+      awesome neovim
+      wordcontainingneovim
+      special
+      specialwithbg
+      specialwithfg
+      ]])
+    feed('Go<tab>neovim tabbed')
+    screen:expect([[
+      {1:neovim}                   |
+      awesome {1:neovim}           |
+      wordcontainingneovim     |
+      special                  |
+      {2:specialwithbg}            |
+      {3:specialwithfg}            |
+                               |
+              {1:neovim} tabbed^    |
+      ~                        |
+      {4:-- INSERT --}             |
+      ]],{
```

Comment:
Is it ok to amend my previous commit and force-push? CONTRIBUTING.md says to do that for [WIP] branches, but this is currently [RFC].